src/App.js

```diff
@@ -1,19 +1,49 @@
 import React from 'react';
+import { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+// import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+// import ChatEntry from './components/ChatEntry';
 
 const App = () => {
+  const [chatMessagesData, setChatMessagesData] = useState(chatMessages);
+
+  const updateChatMessageData = (updatedChat) => {
+    const chats = chatMessagesData.map((chat) => {
+      if (chat.id === updatedChat.id) {
+        return updatedChat;
+      } else {
+        return chat;
+      }
+    });
+    setChatMessagesData(chats);
+  };
+
+  const chatlikes = chatMessagesData.filter((message) => message.liked).length;
+
   return (
     <div id="App">
       <header>
-        <h1>Application title</h1>
+        <h1>Chat between Vladimir and Estragon</h1>
```

Comment:
How can we make this more data driven? Right now we are just hard coding the senders names